### PR TITLE
Make Ubutu and Windows CI flow work again

### PR DESCRIPTION
**Pull request by rebot**
* [x] I have read and followed the guidelines in the `CONTRIBUTING` document
* [x] I have checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.

Implemented the changes I've suggested in #309. CI should also run on this pull request because of the recent change in #316

### CODE REVIEW

**Comment by Roel** [comment]
I think the CI will only kick in once you merge the master branch, so the changed config is applied in this branch too.

**Comment by Roel** [comment]
if we need the additional requirements files for it to work, so be it. We already have a lot of them.. It only needs to be added to the [docs](https://pydov.readthedocs.io/en/stable/development.html#setting-up-your-environment) so we can keep track of things.

By looking at the logs (and execution times) I think the tox commands are never executed now? The check is green, but it never executes the tests. :smile: We might need to add something like [this](https://github.com/ymyzk/tox-gh-actions#factor-conditional-settings-environment-variable) to our tox-gh-actions setup.

I'm also thinking whether the current environments are correct. We now have `{py36,py37,py38,py39}-{lxml,nolxml,windows,others}` which resolves for to 4 environments per Python version: pyXX-lxml, pyXX-nolxml, pyXX-windows and pyXX-others. I think we want pyXX-lxml-windows, pyXX-nolxml-windows, etc., where the -windows and -others variants are skipped depending on the platform? So we might want to change the environment config to something likt `{py36,py37,py38,py39}-{lxml,nolxml}-{windows,others}`?

Lastly, for the conda setup, we might into looking installing the gdal/fiona/geopandas dependencies with conda instead of pip/pipwin?

Thanks for your perseverance, CI is always tricky.. ;)

**Comment by stijnvanhoey** [comment]
I do understand the requirement-file-requirement (although we were trying to decrease the number of them), but I would like to note the inconsistency with what we did earlier in the open PR #309 , see https://github.com/DOV-Vlaanderen/pydov/blob/ce272a91eb2aa0d02de3c3c739b92c8fda976faa/setup.py#L58

I propose to change the name of the `requirements_others.txt` to `requirements_vectorfile.txt` to align things and use the file (similar to the other requirement files) to be read and used, so we get something like:
```
...

with open('requirements_vector.txt') as f:
    requirements_vector = f.read().splitlines()
...
    extras_require={
        'docs': requirements_doc,
        'devs': requirements_dev,
        'vectorfile': requirements_vector
    }
```

File: tox.ini [hunk 1]
```diff
@@ -37,11 +54,9 @@ setenv =
     PYTHONPATH = {toxinidir}
 deps =
     -r{toxinidir}/requirements_dev.txt
-    windows: pipwin
     lxml: lxml
 commands =
-    others: pip install -r {toxinidir}/requirements_win.txt
-    windows: pipwin install -r {toxinidir}/requirements_win.txt
+    pip install -r {toxinidir}/requirements_others.txt
```
Comment (stijnvanhoey) [inline tox.ini:59]: Should this not go to the `deps` section as `-r {toxinidir}/requirements_others.txt`, similar to `-r{toxinidir}/requirements_dev.txt`?

**Comment by stijnvanhoey** [comment]
Note; We have been adding requirement files to the CI just to get the CI up and running. But by defining the dependencies in the `setup.py/setup.cfg` files, installation of the package (but also pytest) will include the required dependencies and install them on the fly and this is also how users will install the pydov package (we probably should not expect they know which requirement files they need to setup a proper virtual environment). 

As such, I do think it make sense to mimic this in the CI as much of possible, e.g. by attempting to get dependencies installed just by doing `pip install .[vectorfile, devs, docs]` instead of first doing `pip install -r ...` (to be honest, I do not know how this would translate to windows). 

Maybe we should merge this, as it brings a solid solution to our windows-CI-issues and I could be wrong as well. But if this comment make sense, we can put it in a separate issue to have a check after merging of this PR and PR #309 ?

**Comment by Roel** [comment]
> As such, I do think it make sense to mimic this in the CI as much of possible, e.g. by attempting to get dependencies installed just by doing `pip install`

I agree, but the issue is precisely that this does not work on Windows. This is why we need to resort to other tools like `pipwin` which is a wrapper that downloads the correct binary wheel from a third-party site and installs that.

If we could just do `pip install` on Windows, all this wouldn't be necessary.

**Comment by pjhaest** [comment]
And what about _conda install_ for windows? @rebot why do you prefer to not use (mini)conda?

**Comment by Roel** [comment]
We have both `pip` and `conda` CI environments, so we need a working setup both with and without conda.

**Comment by pjhaest** [comment]
Ok, but if we distinguish the operating system in the requirements, could we ask for a conda distribution in Windows, or should we keep it open to pip always?

File: tox.ini (hunk 1 above)
Comment (Roel) [reply tox.ini]: Yes, i moved it :)

**Comment by Roel** [comment]
I fixed the tox configuration so the tests are running again, both on Ubuntu and Windows. I'll try Stijn's suggestion with the setup.py next.

For the conda environments, I'd like to install `fiona` and `geopandas` with conda instead of pip(win), but apparently fiona is broken on Python 3.6 on Windows (this is a [known issue](https://github.com/conda-forge/fiona-feedstock/issues/171)) :(

**Review by Roel — APPROVED** [review]
Looks good to me!

I switched the conda tests to install `fiona` and `geopandas` with conda instead of pip(win), unfortunately conda does not support the environment markers in the requirements_vectorfile.txt so I had to specify the packages explicitly.

**Comment by pjhaest** [comment]
Just one question, not the expert in these files: doesn't conda install geopandas also take care of fiona?

**Comment by Roel** [comment]
Yes, since geopandas depends on fiona it will install fiona too. But we rely on fiona directly in GeometryFilter so it's best to add it as a dependency explicitly, as packages can always decide to update their dependencies.

**Comment by Roel** [comment]
I will merge this into the other PR, we can always continue the discussion there. But this will make it easier to compare the changes to master.

**Pull request merged**